ASP/HTML-ASP.sublime-syntax

```diff
@@ -14,9 +14,10 @@ contexts:
         - match: "%>"
           scope: punctuation.section.embedded.end.asp
           pop: true
-        - match: (').*?(?=%>)
-          scope: comment.line.apostrophe.asp
-          captures:
-            1: punctuation.definition.comment.asp
-        - include: scope:source.asp
+        - match: ''
+          push:
+            - include: scope:source.asp # don't directly push, so that the scope doesn't also contain 'source.asp ' as well as our 'source.asp.embedded.html '
+          with_prototype:
+            - match: "(?=%>)"
+              pop: true
```

Comment:
@jcberquist That is a VERY good point. However, if you are defining an ASP function inside of `<%=` inside of a `<script>` tag, and that function switches back to HTML, then… you are going to have other problems. :smile:

I think the distinction here is that is should only re-enter a new sub-context of HTML if it is inside of a nested ASP context like a function. Otherwise I do believe it should pop up to the original HTML syntax.

There may be implementation details I am not thinking of.